sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/WALUtils.scala

```diff
@@ -0,0 +1,280 @@
+/*
```

Comment:
Please leave self-commenting to determine which code blocks are copied from where, and which code is new. And I think this class is to deduplicate between HDFS state store provider and RocksDB state store provider, but HDFS state store provider is not leveraging this, ends up with duplicated code.

Reply:
Good point on self-review. 

I have abstracted out a lot of code from HDFS state store to create WALUtils. I didn't make any change in HDFS state store provider to reduce the scope of this PR. I can either start a new PR for the refactoring or I can do it once the rest of the code is reviewed.